### ISSUE TITLE
localHydroProxy files lost during file migration

### ISSUE THREAD

**Issue by alvacouch** [hydroshare/hydroshare#2040]
@dtarb @hyi @mjstealey @mseul @aphelionz 
When I wrote the `ResourceFile` fix, I didn't account for a category of files for which there are few instances. I accounted for:

* **local** data resources, accessed by the HydroShare proxy user. 
* **federated** resources, accessed via a federation path and federated proxy user. 

I did **not** account for a third category of resources, that are stored on the local **user** server, but accessed through a different proxy account (like `localHydroProxy`). These look a lot like federated resources, but there is one important difference: _Their paths are **dynamically generated** based upon deployment setup, and not fixed._ They also live on our servers rather than on federated servers. 

Thanks to @mjstealey  who just spent an hour with me and helped me get to the bottom of this. 

In short, in the `ResourceFile` makeover, I mistakenly treated these as _federated_ resources, and because of this difference, they became inaccessible. There are about 20 of these. _Fortunately, these resources are easily recognizable via the embedded keyword `localHydroProxy` in their paths._ 

Fix is to treat them differently than federated resources, creating a flag that indicates that their access paths should be _dynamically generated._ Thus, these resources do not have a `resource_federation_path` (which by definition is statically defined), but instead, a Boolean flag that marks them as in the local user proxy zone, from which their paths can be dynamically constructed via setup information.  

Otherwise, their files look like files of local resources.

**Comment by hyi** [comment]
@alvacouch @mjstealey I don't quite understand this third category of resources described and don't quite understand what local user server means in this context. But it looks like deployment-related, so if you all figured out a way to work around or fix this, it is good.

**Comment by alvacouch** [comment]
@hyi @pkdash @mjstealey @dtarb 
See PR #2051 for a hotfix to fix this for 1.10.0. It does nothing in a production environment but it makes testing on alpha, beta, ... work for user-zone files. It implements @hyi's suggestion about rewriting the user zone paths in testing environments. 

The purpose of this PR is to eliminate confusion from broken user-zone resources that aren't really broken in production.

**Comment by alvacouch** [comment]
This issue is completely addressed, but there is more work to be done on avoiding embedding what are essentially configuration constants into file paths. See issue #2104.
